src/renderer/src/App.vue

```diff
@@ -140,6 +151,20 @@ export default defineComponent({
 			(this.$refs.view as any)?.newEditorFromString(file.content);
 		},
 
+		changeTarget(target: Target) {
```

Comment:
Consider passing the target/parser instance directly (via some polymorphic type). Like:

```js
changeParser(parser: MarkupParser) {
   this.parserService = parser;
}
```